pineapple-core/src/main/java/sh/miles/pineapple/command/Command.java

```diff
@@ -53,49 +49,56 @@ public Command(@NotNull final CommandLabel label) {
         this(label, CommandSettings.DEFAULT_COMMAND_SETTINGS);
     }
 
+
     @Override
-    public boolean execute(@NotNull CommandSender sender, @NotNull String[] args) {
-        if (!sender.hasPermission(label.getPermission())) {
-            settings.sendPermissionMessage(sender);
-            return true;
+    public void execute(CommandSourceStack sourceStack, String[] args) {
+        CommandSender executor = sourceStack.getExecutor() != null
+            ? sourceStack.getExecutor()
+            : sourceStack.getSender();
+        if (!executor.hasPermission(label.getPermission())) {
+            settings.sendPermissionMessage(executor);
+            return;
         }
 
         if (args.length == 0) {
-            return noArgExecutor.apply(sender, args);
+            return;
```

Comment:
why did we get rid of the no args executor? It still has utility e.g. for supplying an easy help message etc on commands with sub args